src/build.c

```diff
@@ -817,7 +816,8 @@ static gchar *prepare_run_cmd(GeanyDocument *doc, gchar **working_dir, guint cmd
 	{
 		if (vc->skip_run_script)
 		{
-			utils_free_pointers(2, cmd_string_utf8, working_dir_utf8, NULL);
+			utils_free_pointers(3, cmd_string_utf8, working_dir_utf8,
+				*working_dir, NULL);
```

Comment:
Freeing this breaks https://github.com/geany/geany/blob/a70fd97f9c841e47d4df6363c5870dcd3dd48e97/src/build.c#L876 in the caller

Reply:
Yeah, and as in this code path the function doesn't return `NULL` (or shouldn't, at least), it's expected that `*working_dir` contains the working directory.

I think this change is wrong.  @ntrel did you actually measure a memory leak, or can this be coming from a misinterpretation of the code? (which might suggest it should be made clearer, but well)

Reply:
Freed in the caller here https://github.com/geany/geany/blob/a70fd97f9c841e47d4df6363c5870dcd3dd48e97/src/build.c#L945 I think.

Reply:
Yes

Reply:
Sorry, I messed up here. Replaced this commit with one that just fixes the unused variable.